src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private T[] grow() {
+        return Arrays.copyOf(arrayList, arrayList.length + arrayList.length / 2);
```

Comment:
Let's make this method void(it's private so it's okay) and reassign our array inside